addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/rest/RestClientImpl.java

```diff
@@ -0,0 +1,152 @@
+ */
+package org.openhab.binding.mihome.internal.rest;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.NoSuchAlgorithmException;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.net.ssl.SSLContext;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.InputStreamContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.mihome.internal.ssl.SSLContextBuilder;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the {@link RestClient} interface that is using Jetty HttpClient to perform the requests
+ * and is loading a {@link Certificate} from the OSGi bundle
+ *
+ * @author Svilen Valkanov
+ *
+ */
+public class RestClientImpl implements RestClient {
+
+    private Logger logger = LoggerFactory.getLogger(RestClientImpl.class.getName());
+
+    private HttpClient httpClient;
+    private BundleContext bundleContext;
+    private String baseURL;
+    private int timeout;
+
+    public void activate(BundleContext bundleContext) throws NoSuchAlgorithmException {
+        this.bundleContext = bundleContext;
+
+        httpClient = new HttpClient(new SslContextFactory());
+
+        loadCertificate(CERTIFICATE_DIR, CERTIFICATE_FILE_NAME);
+        setBaseURL(MIHOME_REST_API_URL);
+        setConnectionTimeout(DEFAULT_REQUEST_TIMEOUT);
+        startHttpClient();
+    }
+
+    public void deactivate(BundleContext context) {
+        stopHttpClient();
+    }
+
+    @Override
+    public synchronized ContentResponse sendRequest(String requestPath, HttpMethod httpMethod, Properties httpHeaders,
+            InputStream content, String contentType) throws IOException {
+
+        String url = getBaseURL() + "/" + requestPath;
+        Request request = httpClient.newRequest(url).method(httpMethod).timeout(getConnectionTimeout(),
+                TimeUnit.MILLISECONDS);
+
+        if (httpHeaders != null) {
+            for (String httpHeaderKey : httpHeaders.stringPropertyNames()) {
+                request.header(httpHeaderKey, httpHeaders.getProperty(httpHeaderKey));
+            }
+        }
+
+        // add content if a valid method is given ...
+        if (content != null && (httpMethod.equals(HttpMethod.POST) || httpMethod.equals(HttpMethod.PUT))) {
+            request.content(new InputStreamContentProvider(content), contentType);
+        }
+
+        logger.debug("About to execute {}", request.getURI());
+        ContentResponse response = null;
+        try {
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
```

Comment:
I might have missed this one earlier http://stackoverflow.com/questions/28544867/is-it-ok-to-ignore-interruptedexception-if-nobody-calls-interrupt